### ISSUE TITLE
Passage Authentication - Add configuration parameters for the front-end

### ISSUE THREAD

**Issue by as1729** [usdigitalresponse/cpf-reporter#120]
### Why is this issue important?

See User Story here https://github.com/usdigitalresponse/cpf-reporter/issues/78

This is the first issue in the sequence of tickets to ensure we are able to have a login flow that can be authenticated using https://passage.id/

**Blocked By**
- None

**Blocks**

- [ ] #127 

### Current State

The only way to send the front-end environment-specific variables is through `window.APP_CONFIG.featureFlags` however that is not appropriate for the current use-case.

### Expected State

Developers have the ability to send in environment-specific variables to the front-end that get set in `window.APP_CONFIG.webConfigParams`

### Implementation Plan

Define a new variable that can house the website configuration parameters. This can mimic the same validations of the `website_feature_flags`

```
// terraform/variables.tf

variable "website_config_params" {
  description = "Configurations for the  website that are not feature-specific."
  type        = any
  default     = {}
  validation {
    condition     = can(lookup(var.website_config_params, uuid(), "default"))
    error_message = "Value must be an object."
  }
  validation {
    condition     = can(jsonencode(var.website_config_params))
    error_message = "Value must be JSON-serializable."
  }
}
```

Surface the configuration parameters as options to pass into the `deploy-config.js` file template file.
```
// terraform/web.tf
...
website_config_object_contents = 
  templatefile("${path.module}/tpl/deploy-config.js.tftpl", {
      feature_flags = jsonencode(var.website_feature_flags),
      web_config_params = jsonencode(var.website_config_params),
  })
...
```

Use the newly available `web_config_params` and set the value within `window.APP_CONFIG`
```
// terraform/tpl/deploy-config.js.tftpl
window.APP_CONFIG = window.APP_CONFIG || {};
...
window.APP_CONFIG.webConfigParams = ${web_config_params};
...
```

Ensure Staging environment has the correct `website_configuration_params`
```
// terraform/staging.tfvars

...
// Website
website_domain_name   = "staging.cpf.usdr.dev"
website_feature_flags = {}
website_config_params = {
    passage_app_id = "TBD"
    auth_provider  = "TBD"
}
...
```

**Comment by laurelkorwin** [comment]
Updated context: The changes described in this ticket (adding `website_config_params` into variable and config files similar to `feature_flags`) are done and deployed. However, while attempting to test these changes locally, we realized that `window.APP_CONFIG` is not accessible at all in local OR staging. Per Aditya, in local we'd expect these variables to be accessible via `process.env`, but in staging we would expect that to be accessible even if empty (see default value [here](https://github.com/usdigitalresponse/cpf-reporter/blob/8476d323e597866407f7b3fbb4a8921fd8ab017a/terraform/tpl/deploy-config.js#L1), which should be passed through `web.tf` [here](https://github.com/usdigitalresponse/cpf-reporter/blob/8476d323e597866407f7b3fbb4a8921fd8ab017a/terraform/web.tf#L137)

As part of my PR, I changed the file extension of `deploy-config` in `web.tf` from `.js.tftpl` to `.js` on the hunch that perhaps terraform wasn't able to parse the JS syntax when the file was tagged as tftpl. However, checking in staging, this seems not to have had an effect so we will need to debug further.

Possible thread for investigation is diving into how the terraform configs interact with `redwood.toml` -- see redwood docs [here](https://redwoodjs.com/docs/environment-variables#web).
